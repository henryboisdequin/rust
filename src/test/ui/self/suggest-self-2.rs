struct Foo {}

impl Foo {
    fn foo(&self) {
        bar(self);
        //~^ ERROR cannot find function `bar` in this scope

        bar(&&self, 102);
        //~^ ERROR cannot find function `bar` in this scope

        bar(&mut self, 102, &"str");
        //~^ ERROR cannot find function `bar` in this scope

        bar();
        //~^ ERROR cannot find function `bar` in this scope

        self.bar();
        //~^ ERROR no method named `bar` found for reference `&Foo` in the current scope
    }
}

fn main() {}
