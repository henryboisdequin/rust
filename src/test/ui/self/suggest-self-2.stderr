error[E0425]: cannot find function `bar` in this scope
  --> $DIR/suggest-self-2.rs:5:9
   |
LL |         bar(self);
   |         ^^^

error[E0425]: cannot find function `bar` in this scope
  --> $DIR/suggest-self-2.rs:8:9
   |
LL |         bar(&&self, 102);
   |         ^^^

error[E0425]: cannot find function `bar` in this scope
  --> $DIR/suggest-self-2.rs:11:9
   |
LL |         bar(&mut self, 102, &"str");
   |         ^^^

error[E0425]: cannot find function `bar` in this scope
  --> $DIR/suggest-self-2.rs:14:9
   |
LL |         bar();
   |         ^^^ not found in this scope

error[E0599]: no method named `bar` found for reference `&Foo` in the current scope
  --> $DIR/suggest-self-2.rs:17:14
   |
LL |         self.bar();
   |              ^^^ method not found in `&Foo`

error: aborting due to 5 previous errors

Some errors have detailed explanations: E0425, E0599.
For more information about an error, try `rustc --explain E0425`.
